### ISSUE TITLE
Verify chain / network on RGB invoice

### ISSUE THREAD

**Issue by cryptoquick** [bitmask-stack/bitmask-core#362]
Different from #360; We need to verify if an invoice for the wrong network has been provided, and return an error if that is the case. We'll also need tests, and the necessary upstream functionality.

**Comment by oacdutra** [comment]
Hi @cryptoquick and @josediegorobles , we can make  solve it, if add optional parameter in invoice called network. This maintain compatibility with other wallets and prevents the problem use invoices in wrong network. Does make sense?

**Comment by cryptoquick** [comment]
This should already be on the RGB invoice struct

**Comment by oacdutra** [comment]
> This should already be on the RGB invoice struct

Yes, but it is only used in case you know the beneficiary (see [here](https://github.com/RGB-WG/rgb-wallet/blob/master/src/invoice/parse.rs#L297-L307)).

**Comment by cryptoquick** [comment]
So, that would be an invoice that includes a blinded UTXO? That's fine I think.

**Comment by oacdutra** [comment]
> So, that would be an invoice that includes a blinded UTXO? That's fine I think.

No, RgbInvoice parse the chain only when beneficiary is a bitcoin address, not when a blind UTXO

**Comment by cryptoquick** [comment]
I see. You're right, we may need to add an optional parameter to the invoice then. I'll also tag @dr-orlovsky and @fedsten on this issue so they're aware of the issue.
